internal/server/server.go

```diff
@@ -101,3 +120,45 @@ func New(port int, cert []byte, key []byte, dbPath string, externalHostname stri
 
 	return s, nil
 }
+
+// reconcileCAStatus checks the status of all CAs in the database and updates their status
+// if necessary.
+func ReconcileCAStatus(dbClient *db.Database, logger *zap.Logger) error {
```

Comment:
I understand that you had to make this public to be able to test it, but we should not have to. We should make our `_test.go` files part of the same package, but I tried it locally and at least for this package, it is a relatively big change to the test.

I think we should refactor this in the near future in another PR.

Reply:
I could still write the test under the same package as code here and make the function private if we think that's the right path. The only downside for now is that we'd have a mix of tests being in the same package or in a _test one.